zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -111,25 +146,21 @@ def _test_rmq_amqp_messages_all_units(self, units,
                                                        port=port)
 
                 # Get amqp message
-                logging.info('Get message from:   {} '
+                logging.info('Get messages from:   {} '
                              '({} {})'.format(check_unit_host,
                                               check_unit_name,
                                               check_unit_host_name))
 
-                amqp_msg_rcvd = self._retry_get_amqp_message(check_unit,
-                                                             ssl=ssl,
-                                                             port=port)
-
-                # Validate amqp message content
-                if amqp_msg == amqp_msg_rcvd:
-                    logging.info('Message {} received '
-                                 'OK.'.format(amqp_msg_counter))
-                else:
-                    logging.error('Expected: {}'.format(amqp_msg))
-                    logging.error('Actual:   {}'.format(amqp_msg_rcvd))
-                    msg = 'Message {} mismatch.'.format(amqp_msg_counter)
+                try:
+                    self._search_for_message(
+                        amqp_msg,
+                        check_unit,
+                        ssl,
+                        port,
+                        amqp_msg_counter)
+                except tenacity.RetryError:
```

Comment:
See my other comments: I would expect you'd have to catch an AssertionError instead?